packages/php/compat-checker/src/Checker.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * The Compatibility Checker tool.
+ *
+ * @package Automattic/WooCommerce/Grow/Tools
+ */
+
+namespace Automattic\WooCommerce\Grow\Tools\CompatChecker\v0_0_1;
+
+use Automattic\WooCommerce\Grow\Tools\CompatChecker\v0_0_1\Checks\WPCompatibility;
+use Automattic\WooCommerce\Grow\Tools\CompatChecker\v0_0_1\Checks\WCCompatibility;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * The CompatChecker main class.
+ */
+class Checker {
+
+	/** @var Checker The class instance. */
+	private static $instance;
+
+	/**
+	 * The Plugin instance.
+	 *
+	 * @return Checker
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Parses the plugin contents to retrieve plugin's metadata.
+	 *
+	 * @param string $plugin_file  The Absolute path to the main plugin file.
+	 * @param string $file_version The plugin file version. Can be the same as the plugin version.
+	 */
+	public function get_plugin_data( $plugin_file, $file_version ) {
+		$default_headers = array(
+			'Name'        => 'Plugin Name',
+			'Version'     => 'Version',
+			'RequiresWP'  => 'Requires at least',
+			'RequiresPHP' => 'Requires PHP',
+			'RequiresWC'  => 'WC requires at least',
+			'TestedWP'    => 'Tested up to',
+			'TestedWC'    => 'WC tested up to',
+		);
+
+		$transient_key = 'wc_grow_compat_checker_' . $plugin_file . $file_version;
+		$plugin_data   = get_transient( $transient_key );
+
+		if ( false === $plugin_data ) {
+			$plugin_data = get_file_data( $plugin_file, $default_headers, 'plugin' );
+			set_transient( $transient_key, $plugin_data, MONTH_IN_SECONDS );
```

Comment:
The plugin file version is part of the `$transient_key`. The updated plugin will have a new `$transient_key` because of the version change. (props to @mikkamp for the idea). Discussion about the transient expiration time here: https://github.com/woocommerce/grow/pull/79#issuecomment-1695410469

Reply:
Ah, right! I read through the conversation thread before the review, somehow that part didn't stick  in my brain.